release/models/hashing/openconfig-hashing.yang

```diff
@@ -0,0 +1,238 @@
+module openconfig-hashing {
+    yang-version "1";
+
+    // namespace
+    namespace "http://openconfig.net/yang/hashing";
+
+    prefix "oc-hashing";
+
+    // import some basic types
+    import openconfig-extensions { prefix oc-ext; }
+
+    // meta
+    organization "OpenConfig working group";
+
+    contact
+      "OpenConfig working group
+      netopenconfig@googlegroups.com";
+
+    description
+      "Model for managing hashing policies that would be referenced by the
+      interfaces model.";
+
+
+    oc-ext:openconfig-version "0.1.0";
+
+    revision "2023-08-08" {
+        description "Initial hashing model.";
+        reference "0.1.0";
+    }
+
+    // OpenConfig specific extensions for module metadata.
+    oc-ext:regexp-posix;
+    oc-ext:catalog-organization "openconfig";
+    oc-ext:origin "openconfig";
+
+    typedef hash-field-type-ipv4 {
```

Comment:
For this typedef, please consider:

a) whether this should be an `identity` so that it is more extensible by other modules.
b) providing references as to the possible configurable values across multiple vendors.

Reply:
We have changed the emun to the list of leaf. Please see in https://github.com/openconfig/public/pull/959/files